### PR TITLE
Meterian client runner test

### PR DESCRIPTION
**Pull request by neomatrix369**
Run a the Meterian client runner test in a remote / isolated environment
- test covers a basic flow from setting up client to running it against a vulnerable project and then checking the results of the analysis are as expected (meterian client takes no additional options as params)
- will need authorisation from meterian to allow running analysis against project (plan restrictions apply)
- all basic settings are in place for the meterian client to run
- the necessary assertions are made to verify we have a running client which reports on a vulnerable project 
- added docs to README to describe and clarify usage
- added code coverage generation ability (also amended CircleCI config to store this per build)
- duplications in implementation will be moved to common classes during subsequent tests
- manual tests performed via Jenkins HPI and results are as expected (match to previous runs)

Happy to receive constructive feedback on the changes. Some changes were needed to be able to bring the client under test, although these were mere additions of new constructors or additional methods without impacting original functionality.

### CODE REVIEW

File: src/main/java/com/meterian/common/system/Shell.java [hunk 1]
```diff
@@ -11,6 +11,8 @@
 import java.util.concurrent.Executors;
 import java.util.concurrent.TimeUnit;
 
+import io.meterian.jenkins.core.Meterian;
```
Comment (bbossola) [inline src/main/java/com/meterian/common/system/Shell.java]: This class is now coupled with Meterian. This has to be removed, no coupling should exist between classes in common packages  (com.meterian.common...) with application classes (io.meterian,jenkins)

File: src/main/java/com/meterian/common/system/Shell.java [hunk 2]
```diff
@@ -98,14 +100,30 @@ public String getEnvironmentVariable(String key) {
...
+
+            Meterian.Result result = new Meterian.Result();
+            Shell shell = new Shell();
+
+            Task task = shell.exec(commands, new Options());
+            task.waitFor();
+            result.exitCode = task.exitValue();
+
+            System.out.println(result.exitCode);
+        }
```
Comment (bbossola) [inline src/main/java/com/meterian/common/system/Shell.java]: Move this main in another class. I would suggest moving this main in a test class, even leaving is at a main(). If you want to move it in the TaskTest, I consider the coupling in tests acceptable

File: src/main/java/com/meterian/common/system/Shell.java (hunk 1 above)
Comment (neomatrix369) [reply src/main/java/com/meterian/common/system/Shell.java]: Removed via cab2141

File: src/main/java/com/meterian/common/system/Shell.java (hunk 2 above)
Comment (neomatrix369) [reply src/main/java/com/meterian/common/system/Shell.java]: It was a temporary manual test I wrote, now not necessary anymore

**Pull request merged**